src/vsphere_cpi/lib/cloud/vsphere/resources/vm.rb

```diff
@@ -223,8 +223,9 @@ def power_on
         @client.power_on_vm(datacenter_mob, @mob)
       end
 
-      def upgrade_vm_virtual_hardware
-        @client.upgrade_vm_virtual_hardware(@mob)
+      def upgrade_vm_virtual_hardware(version = nil)
+        version_name = version.nil? ? nil : "vmx-#{version}"
```

Comment:
Seems like it might make sense for

```ruby
version_name = version.nil? ? nil : "vmx-#{version}"
``` 

to be pushed down into  `VCenterClient#upgrade_vm_virtual_hardware` since the conversion here is passed down to that method anyhow...